src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -147,135 +122,176 @@ public void render(GuiGraphics guiGraphics, int mouseX, int mouseY, float partia
 			guiGraphics.renderComponentTooltip(font, tooltip, mouseX, mouseY);
 	}
 	
-	private void drawLadder(GuiGraphics guiGraphics, int currentRung, boolean showLastRung)
+	private int getCurrentRung(int speedFactor)
+	{
+		int currentRung;
+		if(animationCycle == 0)
+		{
+			currentRung = animatedRung;
+			if(animatedRung < ClientPlayerData.getRung())
+			{
+				animatedRungs = ClientPlayerData.getRung() - animatedRung;
+				animationCycle = TIME_BEFORE_ANIMATION + getTicksForRungAnimation(animatedRungs) * speedFactor;
+				animatedRung = ClientPlayerData.getRung();
+			}
+		} else
+		{
+			int rungTicks = getTicksForRungAnimation(animatedRungs) * speedFactor;
+			if(animationCycle - rungTicks >= 0)    //Awaiting animation start
+				currentRung = animatedRung - animatedRungs;
+			else
+			{
+				if(animatedRungs < 5)    //The animation type where the rungs flicker in
+				{
+					int rung = (animationCycle / speedFactor + TIME_BEFORE_NEXT) / (TIME_TILL_NEXT);
+					currentRung = animatedRung - rung;
+					if((animationCycle / speedFactor + TIME_BEFORE_NEXT) % (TIME_TILL_NEXT) >= TIME_BEFORE_NEXT)
+						showLastRung = (animationCycle / speedFactor + TIME_BEFORE_NEXT) % (TIME_TILL_NEXT) - TIME_BEFORE_NEXT >= TIME_FOR_RUNG / 2;
+				} else    //The animation type where the animation just goes through all rungs
+				{
+					int rung = animationCycle * animatedRungs / (TIME_FOR_SHOW_ONLY * speedFactor) + 1;
+					currentRung = animatedRung - rung;
+				}
+			}
+		}
+		return currentRung;
+	}
+	
+	private void drawLadder(GuiGraphics guiGraphics, int currentRung)
 	{
 		RenderSystem.setShaderColor(1, 1, 1, 1);
-		int scroll = scrollIndex % 14;
-		for(int i = 0; i < rows; i++)
-			guiGraphics.blit(guiEcheladder, xOffset + 90, yOffset + 175 + scroll - i*14, 7, 242, 146, 14);
+		int scroll = scrollIndex % RUNG_Y;
 		
-		Random rand = new Random(452619373);
-		for(int i = 0; i < scrollIndex/14; i++)
-			rand.nextInt(0xFFFFFF);
-		for(int i = 0; i < rows; i++)
+		for(int i = 0; i < ROWS; i++)
 		{
-			int y = yOffset + 177 + scroll - i*14;
-			int rung = scrollIndex/14 + i;
-			if(rung > Echeladder.RUNG_COUNT)
+			guiGraphics.blit(guiEcheladder, xOffset + 90, yOffset + 175 + scroll - i * RUNG_Y, 7, 242, 146, RUNG_Y);
+			
+			int y = yOffset + 177 + scroll - i * RUNG_Y;
+			int rung = scrollIndex / RUNG_Y + i;
+			if(rung > (finalRung + 1))
 				break;
 			
-			int textColor = 0xFFFFFF;
+			//TODO broke the animation for previous rung flashing
+			int textColor = currentRung >= rung ? textColor(rung) : 0xFFFFFF;
```

Comment:
Looks like text color no longer takes into account `showLastRung`. This should be fixed.

Reply:
Fixed.